src/lib/encoding.ts

```diff
@@ -0,0 +1,253 @@
+import { Field } from '../snarky';
+
+export {
+  stringToFields,
+  stringFromFields,
+  bytesToFields,
+  bytesFromFields,
+  Bijective,
+};
+
+// functions for encoding data as field elements
+
+// these methods are not for in-snark computation -- from the snark POV,
+// encryption operates on an array of field elements.
+// we also assume here that all fields are constant!
+
+// caveat: this is suitable for encoding arbitrary bytes as fields, but not the other way round
+// to encode fields as bytes in a recoverable way, you need different methods
+
+function stringToFields(message: string) {
+  let bytes = new TextEncoder().encode(message);
+  return bytesToFields(bytes);
+}
+
+function stringFromFields(fields: Field[]) {
+  let bytes = bytesFromFields(fields);
+  return new TextDecoder().decode(bytes);
+}
+
+const STOP = 0x01;
+
+function bytesToFields(bytes: Uint8Array) {
+  // we encode 248 bits (31 bytes) at a time into one field element
+  let fields = [];
+  let currentBigInt = 0n;
+  let bitPosition = 0n;
+  for (let byte of bytes) {
+    currentBigInt += BigInt(byte) << bitPosition;
+    bitPosition += 8n;
+    if (bitPosition === 248n) {
```

Comment:
noob q: why `===` instead of `==` here?

Reply:
JS devs just always use `===` :smile: It's the real equality operator